docs/testing/unit-tests.md

```diff
@@ -1,3 +1,256 @@
+
+## Folder structure
+Keep your tests in a `test` folder in your working directory. The test file should have the same name as the test subject file, and use the extension `.js`.
+
+```
++-- test
+|   +-- bar.js
++-- bar.jsx
+```
+
+Only test files (with at least one test case) should live directly under `/test`. If you need to add external mocks or fixtures, place them in a sub folder, for example:
+
+* `test/mocks/[file-name.js`
+* `test/fixtures/[file-name].js` 
+
+### Importing tests
+
+Given the previous folder structure, try to use relative paths when importing of the __code you're testing__, as opposed to using project paths.
+
+**Good**
+
+`import { bar } from '../bar';`
+
+**Not so good**
+
+`import { bar } from 'components/foo/bar';`
+
+It will make your life easier should you decide to relocate your code to another position in the application directory.
+
+### Describing tests
+Use a 'describe' block to group test cases. Each test case should ideally describe one behaviour only.
+
+In test cases, try to describe in plain words the expected behaviour. For UI components, this might entail describing expected behaviour from a user perspective rather than explaining code internals.
+
+**Good**
+
+```javascript
+describe( 'CheckboxWithLabel', () => {
+    test('checking checkbox should disable the form submit button', () => {
+        ...
+    } );
+} );
+```
+
+**Not so good**
+
+```javascript
+describe( 'CheckboxWithLabel', () => {
+    test('checking checkbox should set this.state.disableButton to `true`', () => {
+        ...
+    } );
+} );
+```
+
+### Snapshot testing
+Snapshot testing is useful for verifying that any data produced during a test is not inadvertently changed. This is especially useful for large and complex data structures like component or state trees. See the [our full reference](snapshot-testing.md) for more information.
+
+### Examples
+
+### Setup and Teardown methods
+
+The Jest API includes some nifty [setup and teardown methods](https://facebook.github.io/jest/docs/en/setup-teardown.html) that allow you to perform tasks *before* and *after* each or all of your tests, or tests within a specific `describe` block.
+
+These methods can handle asynchronous code to allow setup that you normally cannot do inline.
+
+For example, you 
+
+```javascript
+
+// one-time setup for *all* tests
+beforeAll( done => {
+    someAsyncAction()
+        .then( resp => {
+        	window.someGlobal = resp;
+        	done();
+        } );
+} );
+
+// one-time teardown for *all* tests
+afterAll( () => {
+    window.someGlobal = null;
+} );
+
+// one-time setup for *each* test
```

Comment:
We probably can skip examples for `beforeEach` and `afterEach` as they work exactly the same.